tactics.py

```diff
@@ -0,0 +1,108 @@
+denaro = "*"
+settebello_symbol = "07"+denaro
+lay_card_button_name = "Połóż kartę"
+claim_cards_button_name = "Zbierz karty"
+opponent_hand_frame_name = "Ręka przeciwnika"
+actions_frame_name = "Akcje"
+my_hand_frame_name = "Moja ręka"
+table_frame_name = "Stolik"
+this_was_scopa = "To była scopa!"
+
+# factors
+settebello_factor = 5
+card_factor = 1
+seven_factor = 5
+denar_factor = 2
+scopa_factor = 10
+leaving_single_card_factor = -3
+leaving_two_cards_factor = -1
+
+
+def powerset(s):
+    x = len(s)
+    masks = [1 << i for i in range(x)]
+    for i in range(1 << x):
+        yield [ss for mask, ss in zip(masks, s) if i & mask]
+
+
+# returns the list of all possible sums of table, along with cards that are part of this combinat
+# but only if the sum is possible to be taken
+def all_takes_with_sum(s):
+    # store all the sublists
+    ps = powerset(s)
+    takes_with_sum = []
+
+    for cset in ps:
+        take_sum = sum_of_cards(cset)
+        # if the sum is equal or smaller than threshold, append it -
+        # in scopa sums between 8 and 10 as well as anything over 13 is not a valid take
+        if (1 <= take_sum <= 7) or (11 <= take_sum <= 13):
+            takes_with_sum.append([take_sum, cset])
+
+    return takes_with_sum
+
+
+# returns the sum of cards in a list
+def sum_of_cards(clist):
+    val = 0
+    for i in range(0, len(clist)):
+        card_val = clist[i][:2]
+        val = val + int(card_val)
+    return val
+
+
+# return true if settebello
+def settebello(cards):
+    for card in cards:
+        if card == settebello_symbol:
+            return True
+    return False
+
+
+# returns count of sevens
+def sevens(cards):
+    sevens_count = 0
+    for card in cards:
+        if card[1] == "7":
+            sevens_count += 1
+    return sevens_count
+
+
+# returns count of denars
+def denars(cards):
+    denars_count = 0
+    for card in cards:
+        if card[2] == denaro:
+            denars_count += 1
+    return denars_count

```

Comment:
These two functions could be replaced with a single, more generic one in order to avoid code clones.